test/node-api/test_async_context/binding.c

```diff
@@ -0,0 +1,124 @@
+#include <node_api.h>
+#include <assert.h>
+#include "../../js-native-api/common.h"
+
+#define MAX_ARGUMENTS 10
+#define RESERVED_ARGS 3
+
+static napi_value MakeCallback(napi_env env, napi_callback_info info) {
+  size_t argc = MAX_ARGUMENTS;
+  size_t n;
+  napi_value args[MAX_ARGUMENTS];
+  // NOLINTNEXTLINE (readability/null_usage)
+  NAPI_CALL(env, napi_get_cb_info(env, info, &argc, args, NULL, NULL));
+
+  NAPI_ASSERT(env, argc > 0, "Wrong number of arguments");
+
+  napi_value async_context_wrap = args[0];
+  napi_value recv = args[1];
+  napi_value func = args[2];
+
+  napi_value argv[MAX_ARGUMENTS - RESERVED_ARGS];
+  for (n = RESERVED_ARGS; n < argc; n += 1) {
+    argv[n - RESERVED_ARGS] = args[n];
+  }
+
+  napi_valuetype func_type;
+  NAPI_CALL(env, napi_typeof(env, func, &func_type));
+
+  napi_async_context context;
+  NAPI_CALL(env, napi_unwrap(env, async_context_wrap, &context));
+
+  napi_value result;
+  if (func_type == napi_function) {
+    NAPI_CALL(env, napi_make_callback(
+        env, context, recv, func, argc - RESERVED_ARGS, argv, &result));
+  } else {
+    NAPI_ASSERT(env, false, "Unexpected argument type");
+  }
+
+  return result;
+}
+
+static void AsyncDestroyCb(napi_env env, void* data, void* hint) {
+  napi_status status = napi_async_destroy(env, (napi_async_context)data);
+  // We cannot use NAPI_ASSERT_RETURN_VOID because we need to have a JS stack
+  // below in order to use exceptions.
+  assert(status == napi_ok);
+}
+
+static napi_value CreateAsyncResource(napi_env env, napi_callback_info info) {
+  napi_value async_context_wrap;
+  NAPI_CALL(env, napi_create_object(env, &async_context_wrap));
+
+  size_t argc = 2;
+  napi_value args[argc];
```

Comment:
This doesn't compile with MSVC
This should work:
```
  napi_value args[2];
  size_t argc = sizeof(args) / sizeof(args[0]);
```

Reply:
Thanks, I checked out a few of the CI runs and didn't notice those compilation errors. I'll update and re-run the checks.